### ISSUE TITLE
Validation fails if function has no parameters

### ISSUE THREAD

**Issue by werwiesel** [fasseg/exp4j#59]
I have a function without any parameter ("now()"). Expression evaluation is working perfectly, but the validation fails with the error "Too many operators". If I add a parameter to the function call in the expression ("now(0)"), the validation succeeds. I am using version 0.4.5.

**Comment by fasseg** [comment]
@werwiesel: Thanks for the report. I'll have a look at validation.

**Comment by fasseg** [comment]
Added a test and a fix. I sure hope to find time over the next couple of days to roll a new release

**Comment by plap979** [comment]
I think there is an useless if in validate(boolean):
```
if (argsNum > 1) {
    count -= argsNum - 1;
}
```
It can simply be:
```
count -= argsNum - 1;
```
this mighty work in case argsNum == 1 and argsNum == 0 too.

**Comment by fasseg** [comment]
Fixed in exp4j 0.4.6
